no-sdk/java/AWSSignerPostExample.java

```diff
@@ -0,0 +1,152 @@
+                            "x-api-key:" + X_API_KEY + "\n";  // terminate last with \n
+        String signedHeaders = "content-type;host;x-amz-content-sha256;x-amz-date;x-api-key"; 
+
+        String canonicalRequest = METHOD + "\n" +
+                                  canonicalUri + "\n" +
+                                  canonicalQuerystring + "\n" +
+                                  canonicalHeaders + "\n" +
+                                  signedHeaders + "\n" +
+                                  payloadHash;
+
+        String credentialScope = String.format("%s/%s/%s/aws4_request", dateStamp, REGION, SERVICE); 
+
+        String hashedCanonicalRequest = sha256Hex(canonicalRequest);
+        String stringToSign = ALGORITHM + "\n" + 
+                            amzDate + "\n" + 
+                            credentialScope + "\n" + 
+                            hashedCanonicalRequest;
+
+        byte[] signingKey = getSignatureKey(AWS_SECRET_ACCESS_KEY, dateStamp, REGION, SERVICE);
+        String signature = hmacSha256Hex(signingKey, stringToSign);
+
+        // Add signing information to the request
+        String authorizationHeader = String.format("%s Credential=%s/%s, SignedHeaders=%s, Signature=%s", 
+                                    ALGORITHM, AWS_ACCESS_KEY_ID, credentialScope, signedHeaders, signature);
+
+        System.out.println("==========");
+        System.out.println("POSTING " + json + " to " + RESTAPIPATH);
+        System.out.println("authorizationHeader: " + authorizationHeader);
+        System.out.println("==========");
+
+        // Make the request
+        String path = "https://" + RESTAPIHOST + canonicalUri;
+        URL url = new URL(path);
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        con.setDoOutput(true);
+        con.setRequestMethod(METHOD);
+        con.setRequestProperty("content-type", "application/json");
+        con.setRequestProperty("host", RESTAPIHOST);
+        con.setRequestProperty("x-amz-content-sha256", payloadHash);
+        con.setRequestProperty("x-amz-date", amzDate);
+        con.setRequestProperty("x-api-key", X_API_KEY);
+        con.setRequestProperty("Authorization", authorizationHeader);
+        
+        try (OutputStream os = con.getOutputStream()) {
+            byte[] input = json.getBytes("utf-8"); 
+            os.write(input, 0, input.length);			
+        }
+
+        int responseCode = con.getResponseCode();
+        if (responseCode == HttpURLConnection.HTTP_OK) {
+            String responseBody = new String(con.getInputStream().readAllBytes(), StandardCharsets.UTF_8);
+            System.out.println(responseBody);
+        } else {
+            System.out.println("Error: " + responseCode + " " + con.getResponseMessage());
+        }
+    }
+
+    private static byte[] getSignatureKey(String key, String dateStamp, String regionName, String serviceName) throws NoSuchAlgorithmException {
```

Comment:
These helper functions are redundant with AWSSigner.java but I think the redundancy here is worth it for the sake of examples that have everything they need in one file. 

I considered a helper file with shared helper functions, but the fact someone can build the file by itself with a simple `javac AWSSignerPostExample.java` makes the duplication worth while IMO

Reply:
I agree, keeping the examples self-contained is the right decision